### DISCUSSION TITLE
Font assessments

### DISCUSSION THREAD

**Discussion by jelster** [jelster/space-truckers#22] [Ideas]
Quick concept for the main menu to serve as a reference point for discussion 
![image](https://user-images.githubusercontent.com/1508500/106365029-7a684f80-62f8-11eb-8490-ee89fdfec9b6.jpeg).

Note that the final menu contains some items not yet mentioned. Here's what that is going to resemble:

- Main Menu
  -   Start / Resume
  -   How to Play
  -   High Scores
  -   Settings
  -   About

**Comment by eldinor** [comment]
Let's not use italic type in menu.
And, the fonts should be described (and belong to) in Assets.
System fonts could save some download time.

**Reply by jelster** [reply]
The font is called “need for”, I’ll get it into the repos. It’s actually not italic, it’s just the font 😁 I got it from a free game fonts site, I have a couple other candidates too but your point about system fonts rings true - are there enough interesting ones we can assume devices from windows PC’s to iPads, and macs, and android devices will all have?

**Reply by eldinor** [reply]
Defaulting to the system font of a particular operating system can boost performance because the browser doesn’t have to download any font files, it’s using one it already had.

Heres a handy list of font stacks for Window & Mac browsers. For other OS see below.
Arial, Helvetica, Sans-Serif
Arial Black, Gadget, Sans-Serif
Comic Sans MS, Textile, Cursive
Courier New, Courier, Monospace
Georgia, Times New Roman, Times, Serif
Impact, Charcoal, Sans-Serif
Lucida Console, Monaco, Monospace
Lucida Sans Unicode, Lucida Grande, Sans-Serif
Palatino Linotype, Book Antiqua, Palatino, Serif
Tahoma, Geneva, Sans-Serif
Times New Roman, Times, Serif
Trebuchet MS, Helvetica, Sans-Serif
Verdana, Geneva, Sans-Serif
MS Sans Serif, Geneva, Sans-Serif
MS Serif, New York, Serif

https://css-tricks.com/snippets/css/system-font-stack/
https://www.granneman.com/webdev/coding/css/fonts-and-formatting/default-fonts

**Reply by jelster** [reply]
[2-fonts.zip](https://github.com/jelster/space-truckers/files/5898990/2-fonts.zip)

Here's the font along with another one that I thought might be worth a look. The fonts are about 18KB each, so I dunno if font file size will make a huge difference. This is a good list in general though -- I think in the very least we'll want to have fallback fonts and this list will help with that 👍

**Reply by jelster** [reply]
Some other ones I grabbed to evaluate as well
[zero-velocity-brk.zip](https://github.com/jelster/space-truckers/files/5898992/zero-velocity-brk.zip)
[route_159.zip](https://github.com/jelster/space-truckers/files/5898993/route_159.zip)
[gameria.zip](https://github.com/jelster/space-truckers/files/5898994/gameria.zip)

**Reply by eldinor** [reply]
Most of such fonts are quite poorly done from the point of view of professional designer.
Anyway, it is good to have some visual reference, which will help to find more appropriate, properly made fonts.

Here is good article about web font optimization - https://web.dev/optimize-webfont-loading/

Note: You should not store fonts using localStorage or IndexedDB; each of those has its own set of performance issues. The browser's HTTP cache provides the best and most robust mechanism to deliver font resources to the browser.

**Reply by eldinor** [reply]
#### Visual reference for abovementioned fonts

![image](https://user-images.githubusercontent.com/25833327/106371914-56018700-637b-11eb-94fb-dfc2667ce4c0.png)
![image](https://user-images.githubusercontent.com/25833327/106371929-86492580-637b-11eb-9c2e-937bfbefcb42.png)
![image](https://user-images.githubusercontent.com/25833327/106371871-d07dd700-637a-11eb-94dd-53ad647e7180.png)
![image](https://user-images.githubusercontent.com/25833327/106371889-10dd5500-637b-11eb-9aa8-ee68673b1deb.png)
![image](https://user-images.githubusercontent.com/25833327/106371898-30747d80-637b-11eb-96b7-43bae34ae5f2.png)

**Reply by jelster** [reply]
Thanks for posting that comparison - helpful to see the, side by side! I just had another thought regarding fonts. When we talk about Babylon Native and how it doesn't support HTML elements, I think one of the reasons it's so tricky is because Native may not be able to use fonts like a web page does. If that's the case, we would indeed be well served for choosing a font that has a good-enough system font fallback. 

I'll snoop around and see if I can get clarity on this before any decisions have to be made.

**Reply by jelster** [reply]
> #### Visual reference for abovementioned fonts
> 
> 
> 
> ![image](https://user-images.githubusercontent.com/25833327/106371914-56018700-637b-11eb-94fb-dfc2667ce4c0.png)
> 
> ![image](https://user-images.githubusercontent.com/25833327/106371929-86492580-637b-11eb-9c2e-937bfbefcb42.png)
> 
> ![image](https://user-images.githubusercontent.com/25833327/106371871-d07dd700-637a-11eb-94dd-53ad647e7180.png)
> 
> ![image](https://user-images.githubusercontent.com/25833327/106371889-10dd5500-637b-11eb-9aa8-ee68673b1deb.png)
> 
> ![image](https://user-images.githubusercontent.com/25833327/106371898-30747d80-637b-11eb-96b7-43bae34ae5f2.png)
> 
> 

I like the "route 159" font for signage and maybe menu copy. "Need for" would only look good as title or similar type text, if then. Motion control seems promising, but the kerning might make it tougher to read. Time to scour the web for some different fonts!

**Reply by eldinor** [reply]
Probably here is the best solution - https://github.com/jonathantneal/system-font-css
System Font CSS is set of @font-face rules that let you use the native system font of the OS running the browser.

All these fonts which are pictured are unproportional and have a lot of flaws, from professional point of view.
I would suggest to define main fonts later, after we'll understand how to use them in Babylon Native.
Actually we need only 2 fonts: one main and one grotesque.
For road signs we can use PNGs.
